src/stacked_borrows/mod.rs

```diff
@@ -1117,6 +1086,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let val = this.retag_reference(
             &val,
             RefKind::Unique { two_phase: false },
+            RetagKind::Default,
```

Comment:
We should probably have a dedicated retag kind for this, it's a special kind of retag after all.